chainer/sequential.py

```diff
@@ -0,0 +1,400 @@
+        method from ``functools`` package instead::
+
+          from functools import partial
+
+          # This is not pickable
+          model = Sequential(
+              L.Convolution2D(None, 64, 3, 1, 1),
+              lambda x: F.max_pooling_2d(x, 2)
+          )
+
+          # This is pickable
+          model = Sequential(
+              L.Convolution2D(None, 64, 3, 1, 1),
+              partial(F.max_pooling_2d, ksize=2)
+          )
+
+    Args:
+        layers: The layers which are called in its order. Each component should
+            be a callable object including :class:`~Link` object and
+            functions defined under the :mod:`chainer.functions`, e.g.,
+            :obj:`~chainer.functions.relu`, etc.
+
+    """
+
+    def __init__(self, *layers):
+        super(Sequential, self).__init__()
+        self._layers = []
+        for layer in layers:
+            self.append(layer)
+
+    def __len__(self):
+        return len(self._layers)
+
+    def __getitem__(self, i):
+        return self._layers[i]
+
+    def __setitem__(self, i, layer):
+        if i >= len(self):
+            raise ValueError(
+                '{} should be less than {}'.format(i, len(self)))
+        if not callable(layer):
+            raise ValueError(
+                'All elements of a Sequential class should be callable. But '
+                'given {} is not callable.'.format(layer))
+
+        if self._layers[i] is not layer:
+            del self[i]
+            self.insert(i, layer)
+
+    def __delitem__(self, i):
+        layer = self._layers.pop(i)
+        if isinstance(layer, link.Link):
+            for i, _link in enumerate(self._children):
+                if _link is layer:
+                    del self._children[i]
+                    break
+
+    def __iter__(self):
+        return iter(self._layers)
+
+    def __reversed__(self):
+        return reversed(self._layers)
+
+    def __contains__(self, item):
+        return item in self._layers
+
+    def __add__(self, other):
+        if isinstance(other, Sequential):
+            ret = Sequential()
+            for layer in self:
+                ret.append(layer)
+            for layer in other:
+                ret.append(layer)
+            return ret
+        else:
+            raise ValueError('add (+) operator supports only callable '
+                             'objects, but {} is not callable'.format(
+                                 str(type(other))))
+
+    def __radd__(self, other):
+        if callable(other):
+            return Sequential(other) + self
+        else:
+            raise ValueError('add (+) operator is support only callable '
+                             'objects, but {} is not callable'.format(
+                                 str(type(other))))
+
+    def __iadd__(self, other):
+        if isinstance(other, Sequential):
+            for layer in other:
+                self.append(layer)
+        elif callable(other):
```

Comment:
Remove it for consistency with `__add__`

Reply:
Right. Modified.